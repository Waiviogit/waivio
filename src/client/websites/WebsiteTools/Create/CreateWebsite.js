import React, { useCallback, useEffect, useState } from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import PropTypes from 'prop-types';
import { AutoComplete, Button, Checkbox, Input, Form, message } from 'antd';
import { connect } from 'react-redux';
import { debounce, get, isEmpty } from 'lodash';

import validateRules from '../../constants/validateRules';
import { checkAvailableDomain, createNewWebsite, getParentDomainList } from '../../websiteActions';
import { getDomainAvailableStatus, getParentDomain, getWebsiteLoading } from '../../../reducers';

import './CreateWebsite.less';

export const CreateWebsite = ({
  intl,
  form,
  getDomainList,
  parentDomain,
  checkStatusAvailableDomain,
  availableStatus,
  createWebsite,
  loading,
  history,
}) => {
  const { getFieldDecorator, getFieldValue } = form;
  const [searchString, setSearchString] = useState('');
  const [saveWebsite, setWebSiteSave] = useState(false);
  const template = getFieldValue('parent');
  const subDomain = getFieldValue('domain');
  const domainNamesList = Object.keys(parentDomain);
  const showingParentList = searchString
    ? domainNamesList.filter(host => host.includes(searchString))
    : domainNamesList;
  const available = get(availableStatus, 'status');
  const statusMessageClassList = available ? 'CreateWebsite__available' : 'CreateWebsite__error';
  const domainStatus = useCallback(
    debounce(
      () => checkStatusAvailableDomain(getFieldValue('domain'), parentDomain[template]),
      300,
    ),
    [template],
  );

  const handleSearchHost = useCallback(
    debounce(value => setSearchString(value), 300),
    [],
  );

  useEffect(() => {
    getDomainList();
  }, []);

  useEffect(() => {
    if (subDomain) domainStatus();
  }, [template]);

  const handleSubmit = e => {
    e.preventDefault();
    form.validateFieldsAndScroll((err, values) => {
      if (!err && available) {
        createWebsite(values, history)
          .then(() => form.resetFields())
          .catch(error => message.error(error));
        setWebSiteSave(true);
      }
    });
  };

  return (
    <div className="center">
      <h1>
        <FormattedMessage id="create_new_website" defaultMessage="Create new website" />
      </h1>
      <Form className="CreateWebsite" id="CreateWebsite" onSubmit={handleSubmit}>
        <Form.Item>
          <h3>
            <span className="ant-form-item-required">
              {intl.formatMessage({
                id: 'select_website_template',
                defaultMessage: 'Select website template:',
              })}
            </span>
          </h3>
          {getFieldDecorator('parent', {
            rules: validateRules.autocomplete,
          })(<div />)}
          <AutoComplete
            onSelect={value => {
              form.setFieldsValue({ parent: value });
            }}
            onChange={value => handleSearchHost(value)}
          >
            {showingParentList.map(domain => (
              <AutoComplete.Option key={domain} value={domain}>
                {domain}
              </AutoComplete.Option>
            ))}
          </AutoComplete>
        </Form.Item>
        <Form.Item>
          <h3>
            <span className="ant-form-item-required">
              {intl.formatMessage({
                id: 'name_website',
                defaultMessage: 'Specify name for your website:',
              })}
            </span>
          </h3>
          <div className="CreateWebsite__domain-wrap">
            {getFieldDecorator('domain', {
              rules: validateRules.domain,
            })(<Input disabled={!template} id="domain" onInput={domainStatus} />)}
            {template && <span className="CreateWebsite__domain-name">.{template}</span>}
          </div>
        </Form.Item>
        {!isEmpty(availableStatus) && (
          <span className={statusMessageClassList}>{intl.formatMessage(availableStatus.intl)}</span>
        )}
        <p>
          {intl.formatMessage({
            id: 'info_level_domain_name',
            defaultMessage: 'It will be used as a second level domain name.',
          })}
        </p>
        <Form.Item className="CreateWebsite__checkbox-wrap">
          {getFieldDecorator('politics', {
            rules: validateRules.politics,
          })(
            <Checkbox>
              <span className="ant-form-item-required">
                {intl.formatMessage({
                  id: 'website_create_alerts_info',
                  defaultMessage: 'I have read and agree to the terms and conditions of the',
                })}{' '}
                <a
                  href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  {intl.formatMessage({
                    id: 'web_hosting_service',
                    defaultMessage: 'Web Hosting Service Agreement',
                  })}
                </a>
              </span>
            </Checkbox>,
          )}
        </Form.Item>
        <Form.Item>
          <Button type="primary" htmlType="submit" loading={saveWebsite && loading}>
            {intl.formatMessage({
              id: 'create_website',
              defaultMessage: 'Create website',
            })}
          </Button>
        </Form.Item>
      </Form>
    </div>
  );
};

CreateWebsite.propTypes = {
  intl: PropTypes.shape().isRequired,
  form: PropTypes.shape().isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
  getDomainList: PropTypes.func.isRequired,
  createWebsite: PropTypes.func.isRequired,
  parentDomain: PropTypes.arrayOf(PropTypes.string).isRequired,
  checkStatusAvailableDomain: PropTypes.func.isRequired,
  availableStatus: PropTypes.string,
  loading: PropTypes.bool.isRequired,
};

CreateWebsite.defaultProps = {
  availableStatus: '',
};

export default connect(
  state => ({
    parentDomain: getParentDomain(state),
    availableStatus: getDomainAvailableStatus(state),
    loading: getWebsiteLoading(state),
  }),
  {
    getDomainList: getParentDomainList,
    checkStatusAvailableDomain: checkAvailableDomain,
    createWebsite: createNewWebsite,
  },
)(Form.create()(injectIntl(CreateWebsite)));
