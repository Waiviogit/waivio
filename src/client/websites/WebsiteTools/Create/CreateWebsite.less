@import (reference) '../../../styles/custom.less';

.CreateWebsite {
  &__domain-wrap {
    display: flex;

    .ant-input {
      width: 50%;
      margin-right: 5px;
    }
  }

  &__available {
    color: @green;
  }

  &__error {
    color: @red;
  }

  &__domain-name {
    font-weight: bolder;
  }

  .ant-form-explain {
    margin: 0;
    min-height: auto;
  }

  .ant-select-auto-complete.ant-select .ant-input {
    background: @white;
  }

  .ant-form-item-children .ant-checkbox-wrapper {
    display: flex;
    align-items: center;
  }

  .ant-checkbox-wrapper {
    line-height: 1.4;
    margin-top: 25px;
  }

  &__checkbox-wrap.ant-form-item-with-help {
    margin-bottom: 0;

    //.ant-form-item-control.has-success {
    //}
  }
}

@import 'CreateWebsite-nightmode';
