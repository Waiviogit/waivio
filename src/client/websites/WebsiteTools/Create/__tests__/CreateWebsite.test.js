import React from 'react';
import { shallow } from 'enzyme';

import { CreateWebsite } from '../CreateWebsite';

describe('<CreateWebsite />', () => {
  const props = {
    intl: { formatMessage: jest.fn() },
    form: {
      validateFieldsAndScroll: jest.fn(),
      getFieldValue: jest.fn(),
      getFieldDecorator: jest.fn(() => c => c),
    },
    getDomainList: () => {},
    parentDomain: [],
    checkStatusAvailableDomain: jest.fn(),
    availableStatus: { intl: { defaultMessage: 'Available', id: 'available' }, status: 200 },
    createWebsite: jest.fn(),
    loading: false,
  };

  const wrapper = shallow(<CreateWebsite {...props} />);

  it('renders without exploding', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('form submit', () => {
    wrapper.find('#CreateWebsite').simulate('submit', { preventDefault: jest.fn() });
    expect(props.form.validateFieldsAndScroll).toHaveBeenCalled();
  });
});
