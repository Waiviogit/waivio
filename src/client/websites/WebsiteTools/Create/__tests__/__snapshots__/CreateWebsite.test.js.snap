// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateWebsite /> renders without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateWebsite
    availableStatus={
      Object {
        "intl": Object {
          "defaultMessage": "Available",
          "id": "available",
        },
        "status": 200,
      }
    }
    checkStatusAvailableDomain={[MockFunction]}
    createWebsite={[MockFunction]}
    form={
      Object {
        "getFieldDecorator": [MockFunction] {
          "calls": Array [
            Array [
              "parent",
              Object {
                "rules": Array [
                  Object {
                    "message": "Field is required",
                    "required": true,
                  },
                ],
              },
            ],
            Array [
              "domain",
              Object {
                "rules": Array [
                  Object {
                    "message": "Field is required",
                    "required": true,
                  },
                  Object {
                    "pattern": /\\[a-z,0-9\\]\\+\\$/,
                  },
                  Object {
                    "validator": [Function],
                  },
                ],
              },
            ],
            Array [
              "politics",
              Object {
                "rules": Array [
                  Object {
                    "validator": [Function],
                  },
                ],
              },
            ],
          ],
        },
        "getFieldValue": [MockFunction] {
          "calls": Array [
            Array [
              "parent",
            ],
            Array [
              "domain",
            ],
          ],
        },
        "validateFieldsAndScroll": [MockFunction],
      }
    }
    getDomainList={[Function]}
    intl={
      Object {
        "formatMessage": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "defaultMessage": "Select website template:",
                "id": "select_website_template",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Specify name for your website:",
                "id": "name_website",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Available",
                "id": "available",
              },
            ],
            Array [
              Object {
                "defaultMessage": "It will be used as a second level domain name.",
                "id": "info_level_domain_name",
              },
            ],
            Array [
              Object {
                "defaultMessage": "I have read and agree to the terms and conditions of the",
                "id": "website_create_alerts_info",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Web Hosting Service Agreement",
                "id": "web_hosting_service",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Create website",
                "id": "create_website",
              },
            ],
          ],
        },
      }
    }
    loading={false}
    parentDomain={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          <FormattedMessage
            defaultMessage="Create new website"
            id="create_new_website"
            values={Object {}}
          />
        </h1>,
        <Form
          className="CreateWebsite"
          colon={true}
          hideRequiredMark={false}
          id="CreateWebsite"
          layout="horizontal"
          onSubmit={[Function]}
        >
          <FormItem
            hasFeedback={false}
          >
            <h3>
              <span
                className="ant-form-item-required"
              />
            </h3>
            <div />
            <AutoComplete
              choiceTransitionName="zoom"
              filterOption={false}
              onChange={[Function]}
              onSelect={[Function]}
              optionLabelProp="children"
              showSearch={false}
              transitionName="slide-up"
            />
          </FormItem>
          <FormItem
            hasFeedback={false}
          >
            <h3>
              <span
                className="ant-form-item-required"
              />
            </h3>
            <div
              className="CreateWebsite__domain-wrap"
            >
              <Input
                disabled={true}
                id="domain"
                onInput={[Function]}
                type="text"
              />
            </div>
          </FormItem>
          <span
            className="CreateWebsite__available"
          />
          <p />
          <FormItem
            className="CreateWebsite__checkbox-wrap"
            hasFeedback={false}
          >
            <Checkbox
              indeterminate={false}
            >
              <span
                className="ant-form-item-required"
              >
                 
                <a
                  href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                  rel="noopener noreferrer"
                  target="_blank"
                />
              </span>
            </Checkbox>
          </FormItem>
          <FormItem
            hasFeedback={false}
          >
            <Button
              block={false}
              ghost={false}
              htmlType="submit"
              loading={false}
              type="primary"
            />
          </FormItem>
        </Form>,
      ],
      "className": "center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FormattedMessage
            defaultMessage="Create new website"
            id="create_new_website"
            values={Object {}}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultMessage": "Create new website",
            "id": "create_new_website",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <FormItem
              hasFeedback={false}
            >
              <h3>
                <span
                  className="ant-form-item-required"
                />
              </h3>
              <div />
              <AutoComplete
                choiceTransitionName="zoom"
                filterOption={false}
                onChange={[Function]}
                onSelect={[Function]}
                optionLabelProp="children"
                showSearch={false}
                transitionName="slide-up"
              />
            </FormItem>,
            <FormItem
              hasFeedback={false}
            >
              <h3>
                <span
                  className="ant-form-item-required"
                />
              </h3>
              <div
                className="CreateWebsite__domain-wrap"
              >
                <Input
                  disabled={true}
                  id="domain"
                  onInput={[Function]}
                  type="text"
                />
              </div>
            </FormItem>,
            <span
              className="CreateWebsite__available"
            />,
            <p />,
            <FormItem
              className="CreateWebsite__checkbox-wrap"
              hasFeedback={false}
            >
              <Checkbox
                indeterminate={false}
              >
                <span
                  className="ant-form-item-required"
                >
                   
                  <a
                    href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                    rel="noopener noreferrer"
                    target="_blank"
                  />
                </span>
              </Checkbox>
            </FormItem>,
            <FormItem
              hasFeedback={false}
            >
              <Button
                block={false}
                ghost={false}
                htmlType="submit"
                loading={false}
                type="primary"
              />
            </FormItem>,
          ],
          "className": "CreateWebsite",
          "colon": true,
          "hideRequiredMark": false,
          "id": "CreateWebsite",
          "layout": "horizontal",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h3>
                  <span
                    className="ant-form-item-required"
                  />
                </h3>,
                <div />,
                <AutoComplete
                  choiceTransitionName="zoom"
                  filterOption={false}
                  onChange={[Function]}
                  onSelect={[Function]}
                  optionLabelProp="children"
                  showSearch={false}
                  transitionName="slide-up"
                />,
              ],
              "hasFeedback": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="ant-form-item-required"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "ant-form-item-required",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [],
                  "choiceTransitionName": "zoom",
                  "filterOption": false,
                  "onChange": [Function],
                  "onSelect": [Function],
                  "optionLabelProp": "children",
                  "showSearch": false,
                  "transitionName": "slide-up",
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h3>
                  <span
                    className="ant-form-item-required"
                  />
                </h3>,
                <div
                  className="CreateWebsite__domain-wrap"
                >
                  <Input
                    disabled={true}
                    id="domain"
                    onInput={[Function]}
                    type="text"
                  />
                </div>,
              ],
              "hasFeedback": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="ant-form-item-required"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "ant-form-item-required",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Input
                      disabled={true}
                      id="domain"
                      onInput={[Function]}
                      type="text"
                    />,
                    undefined,
                  ],
                  "className": "CreateWebsite__domain-wrap",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "disabled": true,
                      "id": "domain",
                      "onInput": [Function],
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  undefined,
                ],
                "type": "div",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "CreateWebsite__available",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Checkbox
                indeterminate={false}
              >
                <span
                  className="ant-form-item-required"
                >
                   
                  <a
                    href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                    rel="noopener noreferrer"
                    target="_blank"
                  />
                </span>
              </Checkbox>,
              "className": "CreateWebsite__checkbox-wrap",
              "hasFeedback": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span
                  className="ant-form-item-required"
                >
                   
                  <a
                    href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                    rel="noopener noreferrer"
                    target="_blank"
                  />
                </span>,
                "indeterminate": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " ",
                    <a
                      href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                      rel="noopener noreferrer"
                      target="_blank"
                    />,
                  ],
                  "className": "ant-form-item-required",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "href": "https://www.waivio.com/object/snn-web-hosting-agreement/page",
                      "rel": "noopener noreferrer",
                      "target": "_blank",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "a",
                  },
                ],
                "type": "span",
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Button
                block={false}
                ghost={false}
                htmlType="submit"
                loading={false}
                type="primary"
              />,
              "hasFeedback": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "block": false,
                "children": undefined,
                "ghost": false,
                "htmlType": "submit",
                "loading": false,
                "type": "primary",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            <FormattedMessage
              defaultMessage="Create new website"
              id="create_new_website"
              values={Object {}}
            />
          </h1>,
          <Form
            className="CreateWebsite"
            colon={true}
            hideRequiredMark={false}
            id="CreateWebsite"
            layout="horizontal"
            onSubmit={[Function]}
          >
            <FormItem
              hasFeedback={false}
            >
              <h3>
                <span
                  className="ant-form-item-required"
                />
              </h3>
              <div />
              <AutoComplete
                choiceTransitionName="zoom"
                filterOption={false}
                onChange={[Function]}
                onSelect={[Function]}
                optionLabelProp="children"
                showSearch={false}
                transitionName="slide-up"
              />
            </FormItem>
            <FormItem
              hasFeedback={false}
            >
              <h3>
                <span
                  className="ant-form-item-required"
                />
              </h3>
              <div
                className="CreateWebsite__domain-wrap"
              >
                <Input
                  disabled={true}
                  id="domain"
                  onInput={[Function]}
                  type="text"
                />
              </div>
            </FormItem>
            <span
              className="CreateWebsite__available"
            />
            <p />
            <FormItem
              className="CreateWebsite__checkbox-wrap"
              hasFeedback={false}
            >
              <Checkbox
                indeterminate={false}
              >
                <span
                  className="ant-form-item-required"
                >
                   
                  <a
                    href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                    rel="noopener noreferrer"
                    target="_blank"
                  />
                </span>
              </Checkbox>
            </FormItem>
            <FormItem
              hasFeedback={false}
            >
              <Button
                block={false}
                ghost={false}
                htmlType="submit"
                loading={false}
                type="primary"
              />
            </FormItem>
          </Form>,
        ],
        "className": "center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormattedMessage
              defaultMessage="Create new website"
              id="create_new_website"
              values={Object {}}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defaultMessage": "Create new website",
              "id": "create_new_website",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <FormItem
                hasFeedback={false}
              >
                <h3>
                  <span
                    className="ant-form-item-required"
                  />
                </h3>
                <div />
                <AutoComplete
                  choiceTransitionName="zoom"
                  filterOption={false}
                  onChange={[Function]}
                  onSelect={[Function]}
                  optionLabelProp="children"
                  showSearch={false}
                  transitionName="slide-up"
                />
              </FormItem>,
              <FormItem
                hasFeedback={false}
              >
                <h3>
                  <span
                    className="ant-form-item-required"
                  />
                </h3>
                <div
                  className="CreateWebsite__domain-wrap"
                >
                  <Input
                    disabled={true}
                    id="domain"
                    onInput={[Function]}
                    type="text"
                  />
                </div>
              </FormItem>,
              <span
                className="CreateWebsite__available"
              />,
              <p />,
              <FormItem
                className="CreateWebsite__checkbox-wrap"
                hasFeedback={false}
              >
                <Checkbox
                  indeterminate={false}
                >
                  <span
                    className="ant-form-item-required"
                  >
                     
                    <a
                      href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                      rel="noopener noreferrer"
                      target="_blank"
                    />
                  </span>
                </Checkbox>
              </FormItem>,
              <FormItem
                hasFeedback={false}
              >
                <Button
                  block={false}
                  ghost={false}
                  htmlType="submit"
                  loading={false}
                  type="primary"
                />
              </FormItem>,
            ],
            "className": "CreateWebsite",
            "colon": true,
            "hideRequiredMark": false,
            "id": "CreateWebsite",
            "layout": "horizontal",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <h3>
                    <span
                      className="ant-form-item-required"
                    />
                  </h3>,
                  <div />,
                  <AutoComplete
                    choiceTransitionName="zoom"
                    filterOption={false}
                    onChange={[Function]}
                    onSelect={[Function]}
                    optionLabelProp="children"
                    showSearch={false}
                    transitionName="slide-up"
                  />,
                ],
                "hasFeedback": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span
                      className="ant-form-item-required"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "ant-form-item-required",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [],
                    "choiceTransitionName": "zoom",
                    "filterOption": false,
                    "onChange": [Function],
                    "onSelect": [Function],
                    "optionLabelProp": "children",
                    "showSearch": false,
                    "transitionName": "slide-up",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <h3>
                    <span
                      className="ant-form-item-required"
                    />
                  </h3>,
                  <div
                    className="CreateWebsite__domain-wrap"
                  >
                    <Input
                      disabled={true}
                      id="domain"
                      onInput={[Function]}
                      type="text"
                    />
                  </div>,
                ],
                "hasFeedback": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span
                      className="ant-form-item-required"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "ant-form-item-required",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Input
                        disabled={true}
                        id="domain"
                        onInput={[Function]}
                        type="text"
                      />,
                      undefined,
                    ],
                    "className": "CreateWebsite__domain-wrap",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "disabled": true,
                        "id": "domain",
                        "onInput": [Function],
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    undefined,
                  ],
                  "type": "div",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "CreateWebsite__available",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Checkbox
                  indeterminate={false}
                >
                  <span
                    className="ant-form-item-required"
                  >
                     
                    <a
                      href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                      rel="noopener noreferrer"
                      target="_blank"
                    />
                  </span>
                </Checkbox>,
                "className": "CreateWebsite__checkbox-wrap",
                "hasFeedback": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <span
                    className="ant-form-item-required"
                  >
                     
                    <a
                      href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                      rel="noopener noreferrer"
                      target="_blank"
                    />
                  </span>,
                  "indeterminate": false,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      " ",
                      <a
                        href="https://www.waivio.com/object/snn-web-hosting-agreement/page"
                        rel="noopener noreferrer"
                        target="_blank"
                      />,
                    ],
                    "className": "ant-form-item-required",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "href": "https://www.waivio.com/object/snn-web-hosting-agreement/page",
                        "rel": "noopener noreferrer",
                        "target": "_blank",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "a",
                    },
                  ],
                  "type": "span",
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  block={false}
                  ghost={false}
                  htmlType="submit"
                  loading={false}
                  type="primary"
                />,
                "hasFeedback": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "block": false,
                  "children": undefined,
                  "ghost": false,
                  "htmlType": "submit",
                  "loading": false,
                  "type": "primary",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
